src/engine/QueryPlanner.h

```diff
@@ -235,9 +253,10 @@ class QueryPlanner {
   std::vector<SubtreePlan> createExecutionTrees(ParsedQuery& pq,
                                                 bool isSubquery = false);
 
- private:
+ protected:
```

Comment:
Sonar (somewhat rightfully) doesn't like this. 
Make the member private, and write a protected getter function.
And also implement (probably default) the virtual destructor, s.t. we have the tools also satisfied.